scripts/codestyle/cpplint_precommit.hook

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```

Comment:
The precommit hook files are just modified from the paddle repo.

I tried this one https://github.com/bmorcos/pre-commit-hooks-cpp, but got the following error:

```
[INFO] Installing environment for https://github.com/bmorcos/pre-commit-hooks-cpp.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: Command: ('/Users/qijun/.cache/pre-commit/repo4qxgesqb/py_env-python3.7/bin/python3.7', '/Users/qijun/.cache/pre-commit/repo4qxgesqb/py_env-python3.7/bin/pip', 'install', '.')
Return code: 1
Expected return code: 0
Output: (none)
Errors:
    ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

Check the log at /Users/qijun/.cache/pre-commit/pre-commit.log
```

We could use our own scripts currently. After debugging this, we could transfer to pre-made C++ hooks.